src/cascadia/TerminalCore/TerminalDispatch.cpp

```diff
@@ -786,3 +789,84 @@ bool TerminalDispatch::HardReset()
 
     return true;
 }
+
+// Routine Description:
+// - DECSC - Saves the current "cursor state" into a memory buffer. This
+//   includes the cursor position, origin mode, graphic rendition, and
+//   active character set.
+// Arguments:
+// - <none>
+// Return Value:
+// - True.
+bool TerminalDispatch::CursorSaveState()
+{
+    // TODO GH#3849: When de-duplicating this, the AdaptDispatch version of this
+    // is more elaborate.
+    const auto attributes = _terminalApi.GetTextAttributes();
+    COORD coordCursor = _terminalApi.GetCursorPosition();
+    // The cursor is given to us by the API as relative to current viewport top.

```

Comment:
we aren't adjusting for it tho? it sounds relative but line 835 suggests it is absolute

Reply:
I think I'm probably to blame for the confusion of that wording, but the "absolute" mentioned there is in reference to origin more. VT coordinates can either be relative to the margin origin (which I referred to as "relative"), or relative to the screen origin (which I referred to as "absolute"). Just checked the DEC documentation now, and they actually use the terms "displaced" and "absolute", which I guess isn't much better.

Reply:
Ah, thanks. That makes sense :)